src/Chain/Memory/MemoryProviderInterface.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Chain\Memory;
+
+use PhpLlm\LlmChain\Chain\Input;
+
+/**
+ * @author Denis Zunke <denis.zunke@gmail.com>
+ */
+interface MemoryProviderInterface
+{
+    public function loadMemory(Input $input): ?Memory;
```

Comment:
what do you think about having `Memory[]` as return type here? I can imagine use-cases defining multiple ones

Reply:
Mhm. My intention was that the memory providers provide a fully formatted memory string but you may be right that it could also deliver multiple fully formatted memory strings. At least when it is a usecase someone has in mind the possibility is already there and so i will extend it :)